ReactWindows/ReactNative/UIManager/UIViewOperationQueueCollection.cs

```diff
@@ -0,0 +1,564 @@
+        /// </remarks>
+        public bool SynchronouslyUpdateViewOnDispatcherThread(int tag, ReactStylesDiffMap props)
+        {
+            // The native animations module is a single threaded implementation affinitized to the "main" dispatcher thread.
+            // As a result all calls of this method are on main dispatcher thread.
+            // Yet, for secondary views we have to dispatch to correct dispatcher as fast as possible
+            DispatcherHelpers.AssertOnDispatcher();
+
+            UIViewOperationQueueInstance queue = GetQueueByTag(tag, true);
+            if (queue == null)
+            {
+                // Returns false as per the caller's expectation
+                return false;
+            }
+
+            if (queue == MainUIViewOperationQueue)
+            {
+                // Main queue case. Just forward.
+                MainUIViewOperationQueue.NativeViewHierarchyManager.UpdateProperties(tag, props);
+            }
+            else
+            {
+                // Dispatch to the correct thread.
+                DispatcherHelpers.RunOnDispatcher(queue.Dispatcher, CoreDispatcherPriority.High, () =>
+                {
+                    if (queue.NativeViewHierarchyManager.ViewExists(tag))
+                    {
+                        queue.NativeViewHierarchyManager.UpdateProperties(tag, props);
+                    }
+                    else
+                    {
+                        Debug.WriteLine($"View with tag {tag} not found due to race condition");
+                    }
+                });
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Dispatches the view updates.
+        /// </summary>
+        /// <param name="batchId">The batch identifier.</param>
+        internal void DispatchViewUpdates(int batchId)
+        {
+            // Called on layout manager thread
+
+            // Dispatch to all queues
+            foreach (var queue in _dispatcherToOperationQueueInfo.Values)
+            {
+                queue.queueInstance.DispatchViewUpdates(batchId);
+            }
+        }
+
+        private UIViewOperationQueueInstance GetQueueByTag(int tag, bool dontThrow = false)
+        {
+            UIViewOperationQueueInstance queue;
+            if (!_reactTagToOperationQueue.TryGetValue(tag, out queue))
+            {
+                if (dontThrow)
+                {
+                    return null;
+                }
+                else
+                {
+                    throw new InvalidOperationException("No queue for tag " + tag);
+                }
+            }
+            return queue;
+        }
+
+        private void OnViewsDropped(List<int> tags)
+        {
+            if (_uiManagerActionQueue == null)
+            {
+                _uiManagerActionQueue = _reactContext.GetNativeModule<UIManagerModule>().ActionQueue;
+            }
+
+            _uiManagerActionQueue.Dispatch(() =>
+            {
```

Comment:
Any reason to dispatch instead of just locking? Seems like it would be a trade off between having single threaded guarantees, which are nice for reasoning over source code, versus the (presumably minor) tradeoff of stealing the layout thread to do some work.

Reply:
Yeah, didn't want to add locks for this thing + had high hopes on adding that "idle scheduling" for action queues.

---
In reply to: [167537745](https://github.com/Microsoft/react-native-windows/pull/1615#discussion_r167537745) [](ancestors = 167537745)